calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -505,3 +515,40 @@ func (rh resourceHelper) GetTableTemplate(headings []string) (string, error) {
 
 	return buf.String(), nil
 }
+
+// overwriteMetadata compares the Metadata for a ResourceObject and warns if anything was
+// changed that will not be changed as a part of the update. Any discrepancies to the metadata
+// will be changed to match the metadata available in the stored object except for Labels and
+// Annotations.
+func overwriteMetadata(stored, changed ResourceObject) ResourceObject {
+	sm := stored.GetObjectMeta()
+	cm := changed.GetObjectMeta()
+
+	// Set the fields that are allowed to be overwritten (Labels and Annotations)
+	// so that they will not be overwritten.
+	sm.SetAnnotations(cm.GetAnnotations())
+	sm.SetLabels(cm.GetLabels())
+
+	sm.(*v1.ObjectMeta).DeepCopyInto(cm.(*v1.ObjectMeta))
+	return changed
+}
+
+// prepCreateMetadata removes the metadata fields that should not be set from
+// calicoctl. Only the metadata fields Name, Namespace, ResourceVersion, Labels,
+// and Annotations will be kept. All other fields will be set elsewhere if required.
+func prepCreateMetadata(r ResourceObject) ResourceObject {
```

Comment:
Given that these methods are for Update and Create, can we just make that a bit clearer.  How about `mergeMetadataForUpdate` and `prepareMetadataForCreate`

Reply:
Both of these methods serve a related purpose and that is to disallow modifying any of the metadata fields except the Labels and Annotations.  All other fields may be set internally or through other mechanism, but calicoctl will disregard any values supplied through the YAML/JSON files and ensure the existing values for the other other fields is maintained.

I think the comments could do with beefing up with the details.